components/datepicker/stories/datepicker.stories.ts

```diff
@@ -0,0 +1,764 @@
+    await userEvent.click(datepickerInput);
+
+    // No role on the heading and the text is split into `div`s, so had to resort to this
+    expect((await screen.findByShadowText(/November/i)).parentNode).toHaveTextContent(/November 2022/i);
+  },
+};
+
+export const MaxDate: Story = {
+  render: () => html`
+<auro-datepicker
+  maxDate="03/25/2023"
+  setCustomValidityRangeOverflow="Selected date is later than maximum date."
+>
+  <span slot="bib.fullscreen.headline">maxDate Example</span>
+  <span slot="fromLabel">Choose a date</span>
+  <span slot="bib.fullscreen.dateLabel">Choose a date</span>
+</auro-datepicker>
+  `,
+  async play({ canvas, step }) {
+    const datepickerInput = await canvas.findByShadowRole('textbox');
+    await userEvent.click(datepickerInput);
+  },
+};
+
+// TODO: How to translate datepicker setup scripting to template expressions
+//       (https://lit.dev/docs/templates/expressions/#event-listener-expressions),
+//       as in checkboxStories.ResetState?
+export const UpdateMaxDate: Story = {
+  render: () => {
+    return html`
+<auro-datepicker
+  id="maxDateExample"
+  setCustomValidityRangeOverflow="Selected date is later than maximum date."
+>
+  <span slot="bib.fullscreen.headline">maxDate Example</span>
+  <span slot="fromLabel">Choose a date</span>
+  <span slot="bib.fullscreen.dateLabel">Choose a date</span>
+</auro-datepicker>
+<auro-button id="maxDateChange">Change maxDate to 03/18/25</auro-button>
+<auro-button id="resetMaxDate"
+  >Reset Datepicker to Initial Example</auro-button
+>
+
+<script>
+  (function () {
+    const datepicker = document.querySelector('#maxDateExample');
+    const changeMaxDateButton = document.querySelector('#maxDateChange');
+    const resetButton = document.querySelector('#resetMaxDate');
+  
+    const today = formatDateString(new Date());
+  
+    let nextWeek = new Date();
+    let addOneWeek = nextWeek.getDate() + 7;
+  
+    nextWeek.setDate(addOneWeek);
+    const nextWeekStr = formatDateString(nextWeek);
+    
+    datepicker?.setAttribute('value', nextWeekStr);
+    datepicker?.setAttribute('maxDate', nextWeekStr);
+  
+    changeMaxDateButton?.addEventListener('click', () => {
+      datepicker?.setAttribute('maxDate', today);
+    });
+  
+    resetButton?.addEventListener('click', () => {
+      datepicker?.setAttribute('value', nextWeekStr);
+      datepicker?.setAttribute('maxDate', nextWeekStr);
+    });
+
+    function formatDateString(date) {
+      const dd = String("0" + date.getDate()).slice(-2);
+      const mm = String("0" + (date.getMonth() + 1)).slice(-2);
+      const yyyy = date.getFullYear();
+
+      return \`\${mm}/\${dd}/\${yyyy}\`;
+    }
+  })();
+</script>
+  `},
+  parameters: {
+    chromatic: { disableSnapshot: true },
+  },
+  async beforeEach() {
+    MockDate.set('2025-03-18');
+ 
+    // 👇 Reset the Date after each story
+    return () => {
+      MockDate.reset();
+    };
+  },
```

Comment:
Learn more about `beforeEach` in Storybook: https://storybook.js.org/docs/writing-tests/component-testing#run-code-before-each-story-in-a-file

Here, I needed to force a static date so that 1) we can make reasonable assertions in the component test, and/or 2) when Chromatic captures a snapshot, the date isn't changing.